permissions/index.html

```diff
@@ -0,0 +1,198 @@
+          return reg.pushManager.subscribe();
+        }).then(function(reg) {
+          console.log('push reg', reg);
+        });
+      }
+
+      function init() {
+
+        var getPosBtn = document.querySelector('#getPositionBtn');
+        var getNotBtn = document.querySelector('#getNotificationBtn');
+        var getMidBtn = document.querySelector('#getMidiBtn');
+        var getPushNotBtn = document.querySelector('#getPushNotificationBtn');
+
+        // Check for Geolocation API permissions
+        navigator.permissions.query({name:'geolocation'}).then(function(p) {
+          updatePermission('geolocation', p.status);
+
+          p.onchange = function() {
+            updatePermission('geolocation', p.status);
+          };
+        });
+
+        // Check for Notifications API permissions
+        navigator.permissions.query({name:'notifications'}).then(function(p) {
+          updatePermission('notifications', p.status);
+          p.onchange = function() {
+            updatePermission('notifications', p.status);
+          };
+        });
+
+        // Check for Midi/Midi SysEx permissions
+        navigator.permissions.query({name:'midi', sysex:false}).then(function(p) {
+          var update = function() { updatePermission('midi', p.status); };
+          update();
+
+          p.onchange = update;
+        });
+
+        navigator.permissions.query({name:'midi', sysex:true}).then(function(p) {
+          var update = function() { updatePermission('midi-sysex', p.status); };
+          update();
+
+          p.onchange = update;
+        });
+
+        // Check for Push notifications
+        navigator.permissions.query({name:'push', userVisible:true}).then(function(p) {
+          var update = function() { updatePermission('push-notifications', p.status); };
+          update();
+
+          p.onchange = update;
+        });
+
+        getPosBtn.addEventListener('click', function() {
+          navigator.geolocation.getCurrentPosition(function(position) {
+            log('Geolocation permissions granted');
+            log('Latitude:' + position.coords.latitude);
+            log('Longitude:' + position.coords.longitude);
+          });
+        });
+
+        getNotBtn.addEventListener('click', function() {
+          Notification.requestPermission(function(result) {
+            if (result === 'denied') {
+              log('Permission wasn\'t granted. Allow a retry.');
+              return;
+            } else if (result === 'default') {
+              log('The permission request was dismissed.');
+              return;
+            }
+            log('Permission was granted for notifications');
+          });
+        });
+
+        getMidBtn.addEventListener('click', function() {
+          navigator.requestMIDIAccess({sysex:true});
+        });
+
+        getPushNotBtn.addEventListener('click', function() {
+          requestPushPermission();
+        });
+      }
+
+      init();
+    })(document);
```

Comment:
Good point. I should have dropped this earlier. Ditching in the next set of commits.